### PR TITLE
Add ParseACL equality testing by overriding equals method

### PR DESCRIPTION
**Pull request by MBagory**
ACL equality testing wasn't implementing as in JS equals() or objective-C isEqualTo functions

I override equals to compare permissions

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
By analyzing the blame information on this pull request, we identified @grantland to be a potential reviewer.

**Comment by facebook-github-bot-8** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by facebook-github-bot-8** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by grantland** [comment]
I might be missing something, but it doesn't seem like our ObjC SDK fully implements ACL equality checking: 

https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/blob/master/Parse/PFACL.m#L330
https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/blob/master/Parse/Internal/ACL/State/PFACLState.m

PFACL delegates to PFACLState for equality, and PFACLState doesn't implement equality.

**Comment by MBagory** [comment]
PFACLStates seems to further delegate equality to PFBaseState, which compare PFPropertyInfo one by one

In the end it seems to work equivalently, at least in my usecase : I use it to ensure that a user is the only one allowed to read/write an object. I implement it by testing equality with another ACL for this user only

Do you think it could be useful to pull dedicated methods for that ? Like "getExclusiveReadAccessForUser" and getExclusiveWriteAccessForUser"

**Comment by grantland** [comment]
Ah, that makes sense! I don't mind adding private dedicated methods, if that's what you mean.

Also, mind adding tests for this?

**Comment by MBagory** [comment]
I initially though of public methods, but the use case could be easily handled in one line without dedicated methods

Sure for the tests, I'll implement them

File: build.gradle [hunk 1]
```diff
@@ -4,7 +4,11 @@ buildscript {
         mavenCentral()
     }
     dependencies {
+<<<<<<< HEAD
```
Comment (rogerhu) [inline build.gradle]: merge conflict

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,10 @@
-#Thu Nov 06 11:00:14 PST 2014
+#Sun Dec 04 15:25:02 RET 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+<<<<<<< HEAD
```
Comment (rogerhu) [inline gradle/wrapper/gradle-wrapper.properties]: same here

File: Parse/src/main/java/com/parse/ParseACL.java [hunk 3]
```diff
@@ -510,6 +512,46 @@ public void setRoleWriteAccess(ParseRole role, boolean allowed) {
...
+  /**
+   * Get whether an other ParseACL object share the same permissions
+   *
+   * @param other
+   *          The other object to test equality
+   * @return {@code true} if the two objects has same permissions. {@code false} otherwise.
+   *
+   */
+  @Override
+  public boolean equals(Object other) {
```
Comment (rogerhu) [inline Parse/src/main/java/com/parse/ParseACL.java]: test for this stuff?

**Comment by rogerhu** [comment]
Needs rebasing too after a bunch of other merges

File: Parse/src/main/java/com/parse/ParseACL.java (hunk 3 above)
Comment (MBagory) [inline Parse/src/main/java/com/parse/ParseACL.java]: Not yet, but I'm working on it

**Comment by rogerhu** [comment]
Reopen when this is ready... thanks!

**Pull request closed without merging**